forum/qa-plugin/report-reason/script.js

```diff
@@ -29,60 +27,53 @@ document.addEventListener("DOMContentLoaded", () => {
 	    
-        const closer = document.querySelector('.close-preview-btn');
+        const closer = document.querySelector(".close-preview-btn");
         closer.addEventListener("click", () => {
 	    hidePopup();
         });
 
-        const sendButton = document.querySelector('.qa-go-flag-send-button');
+        const sendButton = document.querySelector(".qa-go-flag-send-button");
         sendButton.addEventListener("click", () => {
 
-        const flagReason = document.querySelector("input.qa-spam-reason-radio:checked");
-        const flagNotice = document.querySelector(".qa-spam-reason-text").value;
+            const flagReason = document.querySelector("input.qa-spam-reason-radio:checked");
+            const flagNotice = document.querySelector(".qa-spam-reason-text").value;
 
-        const dataArray = {questionid: flagQuestionid, postid: postId, posttype: postType, reasonid: flagReason.value, notice: flagNotice};
-        const sendData = JSON.stringify(dataArray);
-        const errorMessage = 'Blad serwera. Prosze sprobowac za jakis czas';
-        let errorText = '';
+            const dataArray = {questionid: flagQuestionid, postid: postId, posttype: postType, reasonid: flagReason.value, notice: flagNotice};
+            const sendData = JSON.stringify(dataArray);
+            const errorMessage = "Blad serwera. Prosze sprobowac za jakis czas";
+            let isError = false;
 
-        if(flagReason) {
+            if(flagReason) {
 
-            $.ajax({
-                 type: "POST",
-                 url: flagAjaxURL,
-                 data: { ajaxdata: sendData },
-                 dataType:"json",
-                 cache: false,
-                 success: function(data)
-                 {
-                     if(data.error)
-                     {
-                         errorText = errorMessage;
-                     }
-                     else if(data.success)
-                     {
-                         location.reload();
-                     }
-                     else
-                     {
-	                 errorText = errorMessage;
-                     }
-                 },
-                 error: function(data)
-                 {
-	             errorText = errorMessage;
-                 }
-            });	
-        } else {
-            errorText = "Proszę wybrać powód zgłoszenia!";    
-        }
-        if(errorText)
-        {
-            const errorPopupHtml = document.querySelector('.qa-spam-reason-error');
-	    const errorHtml = document.createElement('p');
-            errorHtml.innerHTML = errorText;
-	    errorPopupHtml.appendChild(errorHtml);
-	}
+                $.ajax({
+                    type: "POST",
+                    url: flagAjaxURL,
+                    data: { ajaxdata: sendData },
+                    dataType:"json",
+                    cache: false,
+                    success: function(data) {
+                        if(data.error) {
```

Comment:
Czy można przyjąć, że jeśli pole `data.success` będzie _falsy_, to AJAX zawiera błąd? Jeśli tak, zapisałbym to w ten sposób:
`if (!data.success) { 
    isError = true;
} else {
    location.reload();
}`